node/src/chain_spec.rs

```diff
@@ -124,16 +136,219 @@ pub fn local_testnet_config() -> Result<ChainSpec, String> {
+				)
+				.into(),
+			),
+			(
+				public_from_ss58::<sr25519::Public>(
+					"2JwCMVvx8DgzpRD7sF1PKpzCDbmGiB2oa67ka2SuUe2TSJgB",
+				)
+				.into(),
+				public_from_ss58::<ed25519::Public>(
+					"2P4Bbk7edF41ny7FSMrQ6u2UsjoTaDhD1DARzwdkeDwBiZfn",
+				)
+				.into(),
+			),
+		]
+	}
+}
+
+struct CrannyKeys;
+
+impl CrannyKeys {
+	fn root() -> AccountId {
+		public_from_ss58::<sr25519::Public>("5CVcJfKKo7uqMGvAE9fzqw66tEfngwJat5FruAsa6hbSkejD")
+			.into()
+	}
+	fn authorities() -> Vec<(AuraId, GrandpaId)> {
+		vec![
+			(
+				public_from_ss58::<sr25519::Public>(
+					"5DDBqKzDw4GnEVmqRXvo8iiWzFxT76E3KUDTk79NnM9F6B8V",
+				)
+				.into(),
+				public_from_ss58::<ed25519::Public>(
+					"5CyuN5TUy6hd1WN2o3uZLpRoBjsAqzXLxUFD2GNT1Sjv3sS5",
+				)
+				.into(),
+			),
+			(
+				public_from_ss58::<sr25519::Public>(
+					"5GhK3Hm39J7yL6ZYoeUxynhfTkPxCd3EqnAPfgHcDo37wqmz",
+				)
+				.into(),
+				public_from_ss58::<ed25519::Public>(
+					"5FBqLTmuJWUFkceoeyWRqrSkYpuqJi9hXNAFFfLL3oTJzSCp",
+				)
+				.into(),
+			),
+			(
+				public_from_ss58::<sr25519::Public>(
+					"5DHwmxfN57NvGpLYFFfxrshnGxccE12VbUGsFCzGSYZQKMfD",
+				)
+				.into(),
+				public_from_ss58::<ed25519::Public>(
+					"5DpXQisSziSLWvRKBPH4F8Twdg89gnKbYpMQDtGmgTJrEzyr",
+				)
+				.into(),
+			),
+		]
+	}
+}
+
+pub fn integritee_mainnet_fresh_config() -> Result<ChainSpec, String> {
+	integritee_chain_spec(
+		"Integritee Mainnet",
+		"integritee-mainnet",
+		GenesisKeys::Integritee,
+		r#"{
+		"ss58Format": 13,
+		"tokenDecimals": 12,
+		"tokenSymbol": "TEER"
+		}"#,
+	)
+}
+
+pub fn cranny_fresh_config() -> Result<ChainSpec, String> {
+	integritee_chain_spec(
+		"Cranny",
+		"cranny",
+		GenesisKeys::Cranny,
+		r#"{
+		"ss58Format": 42,
+		"tokenDecimals": 12,
+		"tokenSymbol": "CRA"
+		}"#,
+	)
+}
+
+pub fn integritee_mainnet_config() -> Result<ChainSpec, String> {
+	ChainSpec::from_json_bytes(&include_bytes!("../res/integritee-mainnet.json")[..])
+}
+
+pub fn cranny_config() -> Result<ChainSpec, String> {
+	ChainSpec::from_json_bytes(&include_bytes!("../res/cranny.json")[..])
+}
+
+pub fn integritee_chain_spec(
+	chain_name: &str,
+	chain_id: &str,
+	genesis_keys: GenesisKeys,
+	token_specs: &str,
+) -> Result<ChainSpec, String> {
+	let (root, endowed, authorities) = match genesis_keys {
+		GenesisKeys::Integritee =>
+			(IntegriteeKeys::root(), vec![IntegriteeKeys::root()], IntegriteeKeys::authorities()),
+		GenesisKeys::Cranny =>
+			(CrannyKeys::root(), vec![CrannyKeys::root()], CrannyKeys::authorities()),
+	};
+
+	// Todo: Chris check wasm binary
+	let wasm_binary = WASM_BINARY.ok_or_else(|| "Development wasm not available".to_string())?;
+
+	Ok(chain_spec(
+		&chain_name,
+		&chain_id,
+		move || {
+			testnet_genesis(wasm_binary, authorities.clone(), root.clone(), endowed.clone(), false)
+		},
+		token_specs,
+	))
+}
+
 /// Configure initial storage state for FRAME modules.
+///
+/// Todo: rename to genesis_config
```

Comment:
why leave a todo here?

Reply:
It was a suggestion from @clangenb , we need to discuss

Reply:
Yes, I think this is not only `testnet` anymore. So it is confusing to call it like that. Generic `genesis_config` is the best name IMO.